library/core/src/mem/maybe_uninit.rs

```diff
@@ -1303,110 +1449,112 @@ impl<T> MaybeUninit<T> {
     /// # Examples
     ///
     /// ```
-    /// #![feature(maybe_uninit_as_bytes, maybe_uninit_slice)]
+    /// #![feature(maybe_uninit_as_bytes, maybe_uninit_write_slice, maybe_uninit_slice)]
     /// use std::mem::MaybeUninit;
     ///
-    /// let val = 0x12345678_i32;
-    /// let uninit = MaybeUninit::new(val);
+    /// let uninit = [MaybeUninit::new(0x1234u16), MaybeUninit::new(0x5678u16)];
     /// let uninit_bytes = uninit.as_bytes();
-    /// let bytes = unsafe { MaybeUninit::slice_assume_init_ref(uninit_bytes) };
-    /// assert_eq!(bytes, val.to_ne_bytes());
+    /// let bytes = unsafe { uninit_bytes.assume_init_ref() };
+    /// let val1 = u16::from_ne_bytes(bytes[0..2].try_into().unwrap());
+    /// let val2 = u16::from_ne_bytes(bytes[2..4].try_into().unwrap());
+    /// assert_eq!(&[val1, val2], &[0x1234u16, 0x5678u16]);
     /// ```
     #[unstable(feature = "maybe_uninit_as_bytes", issue = "93092")]
     pub fn as_bytes(&self) -> &[MaybeUninit<u8>] {
+        let bytes = mem::size_of_val(self);
         // SAFETY: MaybeUninit<u8> is always valid, even for padding bytes
-        unsafe {
-            slice::from_raw_parts(self.as_ptr() as *const MaybeUninit<u8>, mem::size_of::<T>())
-        }
+        unsafe { slice::from_raw_parts(self.as_ptr() as *const MaybeUninit<u8>, bytes) }
```

Comment:
So, my assumption here is that the reason these methods aren't used is because they're inlined, and even a small increase in MIR by calling another method can really bloat the codegen. But I don't actually know how often this method specifically is used, so, I'm fine changing it.

I tried to make sure I just copied the existing implementations wherever possible to reduce codegen differences.

Reply:
Oh, I'm completely wrong here that they're inlined; this is what I get for responding without actually looking at the full code. I'll update to use the method here.